processor/src/main/java/com/linecorp/decaton/processor/runtime/internal/CompletionImpl.java

```diff
@@ -0,0 +1,89 @@
+/*
+ * Copyright 2020 LINE Corporation
+ *
+ * LINE Corporation licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+package com.linecorp.decaton.processor.runtime.internal;
+
+import java.util.Objects;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.CompletionStage;
+import java.util.function.Supplier;
+
+import com.linecorp.decaton.processor.runtime.Completion;
+
+import lombok.Data;
+import lombok.experimental.Accessors;
+
+@Accessors(fluent = true)
+@Data
+public class CompletionImpl implements Completion {
+    private final CompletableFuture<Void> future;
+    private Supplier<TimeoutChoice> expireCallback;
```

Comment:
I think this should be volatile as well, since OffsetReaper thread may refer this while ProcessorThread calls setter.

Reply:
True, gonna fix that.